lib/ansible/module_utils/ce.py

```diff
@@ -61,20 +61,20 @@
     'use_ssl': dict(type='bool'),
     'validate_certs': dict(type='bool'),
     'timeout': dict(type='int'),
-    'transport': dict(choices=['cli']),
+    'transport': dict(default='cli', choices=['cli']),
 }
 ce_argument_spec = {
     'provider': dict(type='dict', options=ce_provider_spec),
 }
 ce_top_spec = {
-    'host': dict(removed_in_version=2.3),
-    'port': dict(removed_in_version=2.3, type='int'),
-    'username': dict(removed_in_version=2.3),
-    'password': dict(removed_in_version=2.3, no_log=True),
-    'use_ssl': dict(removed_in_version=2.3, type='bool'),
-    'validate_certs': dict(removed_in_version=2.3, type='bool'),
-    'timeout': dict(removed_in_version=2.3, type='int'),
-    'transport': dict(choices=['cli']),
+    'host': dict(removed_in_version=2.9),
+    'port': dict(removed_in_version=2.9, type='int'),
+    'username': dict(removed_in_version=2.9),
+    'password': dict(removed_in_version=2.9, no_log=True),
+    'use_ssl': dict(removed_in_version=2.9, type='bool'),
+    'validate_certs': dict(removed_in_version=2.9, type='bool'),
+    'timeout': dict(removed_in_version=2.9, type='int'),
+    'transport': dict(removed_in_version=2.9, choices=['cli']),
```

Comment:
Should this have `default='cli'` as well/

Reply:
`default=cli` is not required for `ce` as it is not used in module code directly (eg: module.params['transport']). The drawback of adding a default at the top level is that for every `ce_*` module run it will display deprecation warning for transport irrespective of transport arg is present in playbook or not.

In case of `nxos` the module code directly references top level transport param (module.params['transport']) that' why it is added as default in top level arg. I will make necessary changes in nxos code to refer transport from provider param (for which it defaults to cli) in a separate PR